### PR TITLE
docs(adr): list ADRs in the READMEs of the FxA packages they impact

### PR DESCRIPTION
**Pull request by jodyheavener**
@LZoog had this [wonderful idea](https://github.com/mozilla/fxa/pull/5213#discussion_r420934566) to update the README of fxa-settings with links to the ADRs that influenced its architecture.

So I ran with it and updated all the package READMEs with their relevant ADRs, and added a little blurb to the root README about why and where they exist.

What do you think? Definitely open to copy changes as well.

### CODE REVIEW

**Comment by jodyheavener** [comment]
Huh. Why do we have a rule that enforces an `email-service-`-prefixed branch name?

File: packages/fxa-content-server/README.md [hunk 1]
```diff
@@ -105,6 +105,18 @@ If you'd like to run only unit tests you can do so in your browser by navigating
 
 ---
 
+## Architectural decisions
+
+The FxA team has made intentional decisions when it comes to the design of this package and its related packages' code bases. Learn more in the following ADRs:
+
+- 0001 - [Isolating payment content with third-party widgets from general account management](https://github.com/mozilla/fxa/blob/master/docs/adr/0001-isolating-payment-content-with-third-party-widgets-from-general-account-management.md)
+- 0002 - [Use React, Redux, and Typescript for subscription management pages](https://github.com/mozilla/fxa/blob/master/docs/adr/0002-use-react-redux-and-typescript-for-subscription-management-pages.md)
```
Comment (LZoog) [inline packages/fxa-content-server/README.md:113]: This one should definitely be moved to `fxa-payments-server`.

File: packages/fxa-auth-server/README.md [hunk 2]
```diff
@@ -221,6 +221,19 @@ manage this config.
...
 
+## Architectural decisions
+
+The FxA team has made intentional decisions when it comes to the design of this package and its related packages' code bases. Learn more in the following ADRs:
+
+- 0003 - [Event Broker for Subscription Platform](https://github.com/mozilla/fxa/blob/master/docs/adr/0003-event-broker-for-subscription-platform.md)
+- 0004 - [Product Capabilities for Subscription Services](https://github.com/mozilla/fxa/blob/master/docs/adr/0004-product-capabilities-for-subscription-services.md)
+- 0005 - [Minimizing password entry](https://github.com/mozilla/fxa/blob/master/docs/adr/0005-minimize-password-entry.md)
+- 0006 - [Utilizing JSON-Schemas, SemVer, and Tooling for JSON Messaging](https://github.com/mozilla/fxa/blob/master/docs/adr/0006-json-schemas-for-messaging.md)
+- 0007 - [Placing subscription info in the fxa-subscriptions claim of JWT access tokens](https://github.com/mozilla/fxa/blob/master/docs/adr/0007-subscription-claim-jwt-access-token.md)
```
Comment (LZoog) [inline packages/fxa-auth-server/README.md:232]: I think this should be set in payments as well.

File: packages/fxa-graphql-api/README.md [hunk 3]
```diff
@@ -19,6 +19,12 @@ npx mocha -r ts-node/register src/test/lib/** -g "bearer token"
 
 Refer to Mocha's [CLI documentation](https://mochajs.org/#command-line-usage) for more advanced test configuration.
 
+## Architectural decisions
+
+The FxA team has made intentional decisions when it comes to the design of this package and its related packages' code bases. Learn more in the following ADRs:
+
+- 0009 - [Consistency in testing tools](https://github.com/mozilla/fxa/blob/master/docs/adr/0009-testing-stacks.md)
```
Comment (LZoog) [inline packages/fxa-graphql-api/README.md:26]: We'll also want "Authentication in Settings Redesign" here.

File: packages/fxa-settings/README.md [hunk 4]
```diff
@@ -67,6 +67,16 @@ _You will eventually be able to view the built Storybook at <http://mozilla.gith
...
 
+## Architectural decisions
+
+The FxA team has made intentional decisions when it comes to the design of this package and its related packages' code bases. Learn more in the following ADRs:
+
+- 0009 - [Consistency in testing tools](https://github.com/mozilla/fxa/blob/master/docs/adr/0009-testing-stacks.md)
+- 0010 - [Transition FxA from Backbone to React](https://github.com/mozilla/fxa/blob/master/docs/adr/0010-transition-fxa-from-backbone-to-react.md)
+- 0011 - [Create a New React Application for the Settings Redesign Project](https://github.com/mozilla/fxa/blob/master/docs/adr/0011-create-new-react-app-for-settings-redesign.md)
+- 0013 - [React Toolchain for Settings Redesign](https://github.com/mozilla/fxa/blob/master/docs/adr/0013-react-toolchain-for-settings-redesign.md)
+- 0014 - [Authentication in Settings Redesign](https://github.com/mozilla/fxa/blob/master/docs/adr/0014-auth-for-settings-redesign.md)
```
Comment (LZoog) [inline packages/fxa-settings/README.md:78]: Also need a link to our CSS ADRs here.

**Comment by dannycoates** [comment]
> Why do we have a CI rule that enforces an `email-service-`-prefixed branch name?

Because building email-service is expensive so we only want to spin up a large instance when it has changes and circleci only has the granularity to do that by branch name.

**Comment by lmorchard** [comment]
Not that I'd block this particular PR over it, but: Might be feasible to automate this? If we added an `Affects: fxa-auth-server, fxa-payments-server` line to ADRs, then a script could manage a chunk of bullet points in the respective READMEs

**Pull request closed without merging**